src/managers/GCommandLoader.js

```diff
@@ -48,23 +49,24 @@ class GCommandLoader {
      * @private
      */
     async __loadCommandFiles() {
-        for await (let file of (await fs.readdirSync(this.cmdDir))) {
-            const fileTypeIndex = (file.lastIndexOf('.') - 1 >>> 0) + 2;
-            const fileName = file.slice(0, fileTypeIndex - 1);
-            const fileType = file.slice(fileTypeIndex);
+        for await (let fsDirent of (await fs.readdirSync(this.cmdDir, { withFileTypes: true }))) {
+            let file = fsDirent.name;
+            // Note: fileName includes the extension (.)
+            const fileName = path.basename(file);
+            const fileType = path.extname(file);
 
-            if (!['js', 'ts'].includes(fileType)) {
+            if (fsDirent.isDirectory()) {
                 await this.__loadCommandCategoryFiles(file);
                 continue;
-            }
+            } else if (!['.js', '.ts'].includes(fileType)) { continue; }
 
             file = await require(`${this.cmdDir}/${file}`);
```

Comment:
```suggestion
            file = require(`${this.cmdDir}/${file}`);
```
Not something added by your PR, but `require` is also not asynchronous and therefore doesn't need to be awaited.